lib/Service/MLP/OptimizerService.php

```diff
@@ -211,8 +214,8 @@ public function optimize(int $maxEpochs,
 			$cost = $this->getAverageCost(
 				$output,
 				...Promise\wait(
-					Promise\all(array_map(function () use ($config, $data, $strategy) {
-						return enqueue(new TrainTask($config, $data, $strategy));
+					Promise\all(array_map(function () use ($newConfig, $collectedData, $strategy) {
+						return enqueue(new TrainTask($newConfig, $collectedData, $strategy));
```

Comment:
this suble change of `$config` to `$newConfig` cost me a few hours to figure out :facepalm: